cmd/backfill/main.go

```diff
@@ -0,0 +1,330 @@
+package main
+
+import (
+	"bytes"
+	"encoding/csv"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/guregu/null"
+
+	"github.com/cmsgov/easi-app/pkg/models"
+)
+
+const (
+	envFile     = "BACKFILL_FILE"
+	envHost     = "BACKFILL_HOST"
+	envAuth     = "BACKFILL_AUTH"
+	envDrop     = "BACKFILL_DROP"
+	healthcheck = "%s/api/v1/healthcheck"
+)
+
+type config struct {
+	file string
+	host string
+	auth string
+}
+
+func main() {
+	cfg := &config{
+		file: os.Getenv(envFile),
+		host: os.Getenv(envHost),
+		auth: os.Getenv(envAuth),
+	}
+	if err := execute(cfg); err != nil {
+		fmt.Fprintf(os.Stdout, "backfill failure: %v\n", err)
+		os.Exit(1)
+	}
+}
+
+func execute(cfg *config) error {
+	url := fmt.Sprintf(healthcheck, cfg.host)
+
+	/* #nosec G107 */
+	if _, err := http.Get(url); err != nil {
+		return fmt.Errorf("failed healthcheck for [%s]: %w", url, err)
+	}
+
+	f, err := os.Open(cfg.file)
+	if err != nil {
+		return err
+	}
+	src := csv.NewReader(f)
+	row, err := src.Read()
+	if err != nil {
+		return fmt.Errorf("failed to read column headers for [%s]: %w", cfg.file, err)
+	}
+	// fmt.Fprintf(os.Stdout, "headers: %v\n", row)
+	_ = row
+	rowCount := 0
+	createCount := 0
+	updateCount := 0
+
+	errs := []error{}
+	for err == nil {
+		row, err = src.Read()
+		if err != nil {
+			if err != io.EOF {
+				errs = append(errs, err)
+			}
+			continue
+		}
+		rowCount++
+		item, rErr := convert(row)
+		if rErr != nil {
+			errs = append(errs, rErr)
+			continue
+		}
+		if item == nil {
+			continue
+		}
+		create, rErr := upload(cfg.host, cfg.auth, item)
+		if rErr != nil {
+			errs = append(errs, rErr)
+		}
+		if create {
+			createCount++
+		} else {
+			updateCount++
+		}
+	}
+
+	fmt.Printf("%d rows processed with %d errors.\n%d creates and %d updates.\n\n", rowCount, len(errs), createCount, updateCount)
+
+	if len(errs) != 0 {
+		fmt.Fprintf(os.Stdout, "problems processing file: %v", errs)
+		return fmt.Errorf("problems processing file: %v", errs)
+	}
+	return nil
+}
+
+func upload(host string, auth string, item *entry) (didCreate bool, err error) {
```

Comment:
In the future it is probably worth returning an enum value here instead of a `didCreate` bool. That way we can communicate other results than just create/update if needed.